ruby/ext/google/protobuf_c/repeated_field.c

```diff
@@ -432,7 +432,7 @@ static VALUE RepeatedField_dup(VALUE _self) {
  */
 VALUE RepeatedField_to_ary(VALUE _self) {
   RepeatedField* self = ruby_to_RepeatedField(_self);
-  int size = upb_Array_Size(self->array);
+  int size = self->array ? upb_Array_Size(self->array) : 0;
```

Comment:
This I'll submit individually, with similar fixes.

Reply:
Did you get a chance to submit a separate PR for this? I didn't see one.

What was the root cause that led you to this fix? I ask because it looks very similar to what I've been debugging in https://github.com/protocolbuffers/protobuf/issues/11968

There is a similar symptom; `self->array` ends up being `NULL`. But in that case it's happening in `RepeatedField_each`. I've isolated it to a race condition involving the object cache.

Reply:
Hum, I was working on it at some point but never submitted it. I think at some point I assumed this case wasn't supposed to happen and was just a fallout of another bug, but I might have been wrong not sure.

I can submit a PR if you want.

Reply:
I don't think this case is supposed to happen, and I do think it is the fallout of another bug. I think this is probably not the right fix.

But I am curious what triggered this bug for you. I'm wondering if it's a similar root cause to https://github.com/protocolbuffers/protobuf/issues/11968 (a race condition on the object cache) or something else.